var/spack/repos/builtin/packages/tau/package.py

```diff
@@ -128,8 +128,9 @@ class Tau(Package):
     depends_on("binutils+libiberty+headers+plugins", when="+binutils")
     # Build errors with Python 3.9
     depends_on("python@2.7:3.8", when="@:2.31.0+python")
-    # python 3.11 doesn't work as of 2.32
-    depends_on("python@2.7:3.10", when="@2.31.1:+python")
+    # python 3.11 doesn't work in the 2.32 releases
+    depends_on("python@2.7:3.10", when="@2.31.1:2.32.1+python")
+    depends_on("python@2.7:", when="@2.33:+python")
```

Comment:
I would leave the forward compat bounds separate from the backward compat bounds so that every `depends_on` statement is minimal:

```
with when("+python"):
    depends_on("python@2.7:", when="+python")  # actually `@2.7:` is redundant since we don't support <3 anyways.

    # forward compat bounds:
    depends_on("python@:3.8", when="@:2.31.0")
    depends_on("python@:3.10", when="@:2.32.1")
```